diff --git a/src/main/scala/amba/axi4/Parameters.scala b/src/main/scala/amba/axi4/Parameters.scala
index 088c226d8..20522d11e 100644
--- a/src/main/scala/amba/axi4/Parameters.scala
+++ b/src/main/scala/amba/axi4/Parameters.scala
@@ -127,10 +127,10 @@ case class AXI4BundleParameters(
   val lenBits   = AXI4Parameters.lenBits
   val sizeBits  = AXI4Parameters.sizeBits
   val burstBits = AXI4Parameters.burstBits
-  val lockBits  = AXI4Parameters.lockBits
-  val cacheBits = AXI4Parameters.cacheBits
-  val protBits  = AXI4Parameters.protBits
-  val qosBits   = AXI4Parameters.qosBits
+  val lockBits  = 0 //AXI4Parameters.lockBits
+  val cacheBits = 0 //AXI4Parameters.cacheBits
+  val protBits  = 0 //AXI4Parameters.protBits
+  val qosBits   = 0 //AXI4Parameters.qosBits
   val respBits  = AXI4Parameters.respBits
 
   def union(x: AXI4BundleParameters) =
diff --git a/src/main/scala/util/RationalCrossing.scala b/src/main/scala/util/RationalCrossing.scala
index 3ad9b12c3..180680848 100644
--- a/src/main/scala/util/RationalCrossing.scala
+++ b/src/main/scala/util/RationalCrossing.scala
@@ -153,7 +153,7 @@ class RationalCrossingFull[T <: Data](gen: T, direction: RationalDirection = Sym
 
 object ToRational
 {
-  def apply[T <: Data](x: DecoupledIO[T], direction: RationalDirection = Symmetric): RationalIO[T] = {
+  def apply[T <: Data](x: ReadyValidIO[T], direction: RationalDirection = Symmetric): RationalIO[T] = {
     val source = Module(new RationalCrossingSource(chiselTypeOf(x.bits), direction))
     source.io.enq <> x
     source.io.deq
@@ -162,7 +162,7 @@ object ToRational
 
 object FromRational
 {
-  def apply[T <: Data](x: RationalIO[T], direction: RationalDirection = Symmetric): DecoupledIO[T] = {
+  def apply[T <: Data](x: RationalIO[T], direction: RationalDirection = Symmetric): ReadyValidIO[T] = {
     val sink = Module(new RationalCrossingSink(chiselTypeOf(x.bits0), direction))
     sink.io.enq <> x
     sink.io.deq
